src/CMakeLists.txt

```diff
@@ -78,17 +72,12 @@ if (XGBOOST_BUILTIN_PREFETCH_PRESENT)
     -DXGBOOST_BUILTIN_PREFETCH_PRESENT=1)
 endif (XGBOOST_BUILTIN_PREFETCH_PRESENT)
 
-if (USE_OPENMP)
+if (USE_OPENMP OR USE_CUDA)  # CUDA requires OpenMP
   find_package(OpenMP REQUIRED)
-  if (OpenMP_CXX_FOUND OR OPENMP_FOUND)
-    target_compile_options(objxgboost PRIVATE  $<$<COMPILE_LANGUAGE:CXX>:${OpenMP_CXX_FLAGS}>)
-    if ((NOT OpenMP_CXX_LIBRARIES) AND (NOT MSVC))  # old CMake doesn't define this variable
-      set(OpenMP_CXX_LIBRARIES "gomp;pthread")
-    endif ((NOT OpenMP_CXX_LIBRARIES) AND (NOT MSVC))
-    list(APPEND SRC_LIBS ${OpenMP_CXX_LIBRARIES})
-    set(LINKED_LIBRARIES_PRIVATE "${LINKED_LIBRARIES_PRIVATE};${SRC_LIBS}" PARENT_SCOPE)
-  endif (OpenMP_CXX_FOUND OR OPENMP_FOUND)
-endif (USE_OPENMP)
+  list(APPEND SRC_LIBS OpenMP::OpenMP_CXX)
+  set(LINKED_LIBRARIES_PRIVATE "${LINKED_LIBRARIES_PRIVATE};${SRC_LIBS}" PARENT_SCOPE)
+  target_link_libraries(objxgboost PUBLIC OpenMP::OpenMP_CXX)
```

Comment:
Note: This line is only permissible for CMake 3.12+. See https://gitlab.kitware.com/cmake/cmake/issues/18010

Reply:
Does this PUBLIC linking propagate to user of XGBoost?

Reply:
Changed to PRIVATE.